hooks/src/lib/type.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * UseInput : 단일 input 태그의 입력값을 제어하고, handleChange, handleBlur 이벤트 핸들러와 에러 상태 반환
+ */
+export interface UseInput {
+  value: string;
+  setValue: React.Dispatch<React.SetStateAction<string>>;
+  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+/**
+ * UseCard : 카드 등록 도메인에 대한 단일 input 커스텀 훅 반환 타입.
+ * - value: 입력 필드의 현재 값.
+ * - handleChange: 입력 필드의 값이 변경될 때 호출되는 함수. 입력 값의 유효성을 실시간으로 검사하고, 유효하지 않은 값에 대해서는 상태 업데이트 중단
+ * - handleBlur: 입력 필드에서 포커스가 벗어날 때 호출되는 함수. 이 함수는 최종 값의 형식 검증.
+ * - errorInfo: 현재 입력 값의 유효성 정보를 담고 있는 객체. isValid와 errorMessage 속성을 포함.
+ */
+export interface UseCard extends Omit<UseInput, 'setValue'> {
+  handleBlur: (event: React.FocusEvent<HTMLInputElement, Element>) => void;
+  errorInfo: ValidationResult;
+}
```

Comment:
- 인터페이스에 대한 설명을 주석으로 작성하기 전에 해당 설명을 좀 더 이름에 녹여서 이름만으로 설명이 되게 처리할 수는 없을지 고민해봤으면 좋겠습니다!
- 각 props에 대해 보충 설명이 필요한 경우 line by line으로 적어주시는 것이 찾기에도 쉽고 props 제거시 한번에 제거가 가능할 듯 싶네요!

```suggestion
export interface UseCardInputResult extends Omit<UseInput, 'setValue'> {
  /** 설명 주석 */
  handleBlur: (event: React.FocusEvent<HTMLInputElement, Element>) => void;
  /** 이미 정의된 인터페이스, 타입을 차용했다면 주석 스킵 */
  errorInfo: ValidationResult;
}
```